ext/bcmath/libbcmath/src/str2num.c

```diff
@@ -143,6 +143,7 @@ bool bc_str2num(bc_num *num, const char *str, const char *end, size_t scale, boo
 		fractional_ptr = fractional_end = decimal_point + 1;
 		/* For strings that end with a decimal point, such as "012." */
 		if (UNEXPECTED(*fractional_ptr == '\0')) {
+			*full_scale = 0;
```

Comment:
There's an inconsistency here: why do you check if `full_scale` is NULL down below but not here?

Reply:
Fixed in https://github.com/php/php-src/pull/13741/commits/3c15352385e059bc26c44639f6477d9a6acbe027
I forgot the check here